### PR TITLE
TST: Use tmp_path fixture for temporary files in unit tests

### LINKED ISSUE #466: FITS file deletion fails when tests fail
When unit tests that create output FITS files fail, the test cleanup code might try to delete nonexisting files, causing other errors that are unrelated to what's actually failing in the test.

Instead of using the cleanup function, we can use a pytest `tmp_path` fixture to write files to a pytest temporary directory.

From their [tutorial](https://docs.pytest.org/en/latest/how-to/tmpdir.html):

```py
def mytest(tmp_path):
    out_path = tmp_path / 'test.fits'
```

### PR DESCRIPTION
**Pull request by rrjbca**
## Description
Use pytest `tmp_path` fixture in unit tests. Creates a temporary directory where temporary files can be written and handles cleanup automatically. Resolves #466 

## Checklist
- [x] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/main/CONTRIBUTING.rst)
- [ ] Write unit tests
- [ ] Write documentation strings
- [ ] Assign someone from your working team to review this pull request
- [x] Assign someone from the infrastructure team to review this pull request

### CODE REVIEW

File: skypy/pipeline/tests/test_skypy.py [hunk 1]
```diff
@@ -36,36 +35,39 @@ def test_skypy():
     assert e.value.code == 2
 
     # Process empty config file
-    filename = get_pkg_data_filename('data/empty_config.yml')
-    assert skypy.main([filename, 'empty.fits']) == 0
+    config_filename = get_pkg_data_filename('data/empty_config.yml')
```
Comment (Lucia-Fonseca) [inline skypy/pipeline/tests/test_skypy.py:38]: why don't we use a temporary YAML file here too?

**Review by Lucia-Fonseca — COMMENTED** [review]
The code fixes the problem.
However, I wonder whether `unittest.mock` is more suitable for mocking files, so we don't need to create temporary files or have empty yml files(?)

File: skypy/pipeline/tests/test_pipeline.py [hunk 2]
```diff
@@ -44,7 +43,7 @@ def test_pipeline():
 
     pipeline = Pipeline(config)
     pipeline.execute()
-    output_filename = 'output.fits'
+    output_filename = str(tmp_path / 'output.fits')
```
Comment (philipp128) [inline skypy/pipeline/tests/test_pipeline.py:46]: Do we need the `str` command? It is not used in the [tutorial](https://docs.pytest.org/en/6.2.x/tmpdir.html)

Comment (rrjbca) [reply skypy/pipeline/tests/test_pipeline.py:46]: "`tmp_path` is a `pathlib.Path` object" but our script takes path names as strings

**Comment by rrjbca** [comment]
> However, I wonder whether `unittest.mock` is more suitable for mocking files, so we don't need to create temporary files or have empty yml files(?)

This PR is not mocking any files. It is implementing a more robust method (a temporary directory) to automatically handle the deletion of files created by the skypy script during testing. We don't want to mock those files; the fundamental purpose of these unit tests is to check that those files are created correctly. Mocking config files might be possible but is unrelated to the issue this PR is addressing and I don't see an obvious benefit?

File: skypy/pipeline/tests/test_skypy.py (hunk 1 above)
Comment (rrjbca) [reply skypy/pipeline/tests/test_skypy.py:38]: This is a (non-temporary) input file that is part of the repository

**Review by Lucia-Fonseca — APPROVED** [review]

**Pull request merged**